### ISSUE TITLE
Add CODEOWNERS file to identify maintainers for Eiffel projects

### ISSUE THREAD

**Issue by fdegir** [eiffel-community/community#7]
As an Eiffel Community Member, I would like the project maintainers identified so that I know who to talk to when it comes to project/technical topics.

**Comment by SantoshNC68** [comment]
We have the maintainer groups available, is it to document those changes?
If you need more info about the maintainers here is a list of all the maintainers specific to projects in this community https://github.com/orgs/eiffel-community/teams

**Comment by fdegir** [comment]
yes, it is mainly documenting who are the maintainers of all the projects. for the projects without named maintainers, they should be identified and documented.

general practice followed by the open source projects is to make the list of maintainers visible within the project repositories by adding files in certain format in them, listing the maintainers so people do not have to browse through various places in order to find who should they reach out for different reasons.

one of the ideas which will be proposed is to start using Github CODEOWNERS files which can contain the list of maintainers. in addition to listing maintainers in them, these files enable Github to notify those people automatically for PRs.

https://help.github.com/en/articles/about-code-owners

**Comment by e-pettersson-ericsson** [comment]
I really like the idea of having a CODEOWNERS file in each of the Eiffel repositories.

I also noticed on https://eiffel-community.github.io/ that the section **Getting Involved** stated: 
"_Every repository in the Eiffel community maintains an issue list, accepts pull requests and features a maintainers' mailing list (linked from each respository README)._"

Since I haven't seen any contact information visible in any of the repositories I think this should be addressed properly. Can we start with the community repository to set a template, and then trickle down the list of repositories by adding issues in them too?

**Comment by e-pettersson-ericsson** [comment]
I noticed the CONTACT.md only contains a list of email adresses, but I was looking for something like a bullet list in the README in each of the repositories providing the \@github-usernames of the current maintainers, so it's easier to come into contact with someone directly.

Also that Eiffel is on Slack is not visible at all - I'll create a separate issue for that.

**Comment by e-pettersson-ericsson** [comment]
> one of the ideas which will be proposed is to start using Github CODEOWNERS files which can contain the list of maintainers. in addition to listing maintainers in them, these files enable Github to notify those people automatically for PRs.

So I see two options suggested:

1. Write down maintainer members in the README (names + github handles would suffice I guess)
2. Add a CODEOWNERS file with all the maintainers

The second option would have the added benefit of automatically requesting reviews from the maintainers whenever a new PR is opened to the repository (without having to "watch" the repository I assume?).

@eiffel-community/eiffel-community-maintainers  any thougths regarding the two proposed options?

**Comment by e-backmark-ericsson** [comment]
I like the idea of documenting the maintainers through the CODEOWNERS file. It would keep the readme file cleaner, and the maintainers would automatically get notified about PRs.
Is there a possibility to sync a CODEOWNERS file with the 'maintainers team group' that we currently have set up for each repository (more or less at least). Else I see a risk that these two diverge.

**Comment by e-pettersson-ericsson** [comment]
I read through the guide again https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax and it seems possible to put an entire team (for example the maintainers group) so they should be able to stay synced, if we don't want to use specific usernames.

**Comment by henning-roos** [comment]
I am reopening this issue since the following eiffel repositories do not have a CODEOWNERS file:
https://github.com/eiffel-community/eiffel-graphql-api
https://github.com/eiffel-community/eiffel-remrem-publish
https://github.com/eiffel-community/eiffel-remrem-generate
https://github.com/eiffel-community/eiffel-store
https://github.com/eiffel-community/eiffel-intelligence
https://github.com/eiffel-community/eiffel-easy2use
https://github.com/eiffel-community/eiffel-gerrit-lib
https://github.com/eiffel-community/eiffel-remrem-semantics
https://github.com/eiffel-community/eiffel-remrem-protocol-interface
https://github.com/eiffel-community/eiffel-gerrit-herald
https://github.com/eiffel-community/eiffel-pythonlib
https://github.com/eiffel-community/eiffel
https://github.com/eiffel-community/eiffel-persistence-technology-evaluation
https://github.com/eiffel-community/eiffel-remrem
https://github.com/eiffel-community/eiffel-remrem-parent
https://github.com/eiffel-community/eiffel-remrem-shared
https://github.com/eiffel-community/eiffel-event-repository
https://github.com/eiffel-community/eiffel-jenkins-plugin
https://github.com/eiffel-community/eiffel-gerrit-plugin
https://github.com/eiffel-community/eiffel-intelligence-frontend
https://github.com/eiffel-community/eiffel-sepia
https://github.com/eiffel-community/eiffel-vici
https://github.com/eiffel-community/eiffel-community.github.io
https://github.com/eiffel-community/eiffel-commons
https://github.com/eiffel-community/simple-event-sender
https://github.com/eiffel-community/eiffelactory
https://github.com/eiffel-community/ml-jmespath-generator
https://github.com/eiffel-community/eiffel-jira-plugin
